contracts/helpers/BaseHarvester.sol

```diff
@@ -2,201 +2,280 @@
 
 pragma solidity ^0.8.23;
 
-import { SafeCast } from "oz/utils/math/SafeCast.sol";
-
-import { ITransmuter } from "interfaces/ITransmuter.sol";
-
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { AccessControl, IAccessControlManager } from "../utils/AccessControl.sol";
-import "../utils/Constants.sol";
-import "../utils/Errors.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ITransmuter } from "../interfaces/ITransmuter.sol";
+import { IAgToken } from "../interfaces/IAgToken.sol";
 
-import { IRebalancerFlashloan } from "../interfaces/IRebalancerFlashloan.sol";
+import "../utils/Errors.sol";
+import "../interfaces/IHarvester.sol";
 
-struct CollatParams {
-    // Yield bearing asset associated to the collateral
-    address asset;
-    // Target exposure to the collateral asset used
+struct YieldBearingParams {
+    // Address of the stablecoin (ex: USDC)
+    address stablecoin;
+    // Target exposure to the collateral yield bearing asset used
```

Comment:
Will this always be a stablecoin? Maybe consider name it `depositAsset` or something.